presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkRddFactory.java

```diff
@@ -262,9 +266,13 @@ private PrestoSparkTaskSourceRdd createTaskSourcesRdd(
             PartitioningHandle partitioning,
             List<TableScanNode> tableScans,
             Map<PlanNodeId, SplitSource> splitSources,
-            Optional<Integer> numberOfShufflePartitions)
+            Optional<Integer> numberOfShufflePartitions,
+            Optional<NativeExecutionNode> nativeExecution)
     {
         ListMultimap<Integer, SerializedPrestoSparkTaskSource> taskSourcesMap = ArrayListMultimap.create();
+        boolean nativeExecutionEnabled = isNativeExecutionEnabled(session);
```

Comment:
looks like we only need the nativeExecutionNode here for the sourceId.  Can we instead pull this check out to when we extract the nativeExecutionNode, and pass just the sourceId into this method?